src/App.tsx

```diff
@@ -1,24 +1,191 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { TodoList } from './components/TodoList/TodoList';
+import {
+  deleteTodo,
+  getTodos,
+  postTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Errors } from './types/Errors';
+import { Footer } from './components/Footer/Footer';
+import { FilterType } from './types/FilterType';
+import { filterTodos } from './utils/helpers';
+import { ErrorNotify } from './components/ErrorNotify/ErrorNotify';
+import { Header } from './components/Header/Header';
 
-const USER_ID = 0;
+const USER_ID = 12041;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<Errors | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [selectedTodos, setSelectedTodos] = useState<number[]>([0]);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(event.target.value);
+  };
+
+  const handleError = (err: Errors) => {
+    setError(err);
+
+    setTimeout(() => {
+      setError(null);
+    }, 3000);
+  };
+
+  const handleDelete = async (todoId: number) => {
+    try {
+      if (error) {
+        setError(null);
+      }
+
+      setSelectedTodos(prev => [...prev, todoId]);
+
+      await deleteTodo(todoId);
+
+      setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+    } catch (e) {
+      handleError(Errors.UnableToDelete);
+    } finally {
+      setSelectedTodos([]);
```

Comment:
But what if some other API requests in process and you still need to show loaders for them?